sklearn/utils/estimator_checks.py

```diff
@@ -1135,11 +1181,12 @@ def check_classifiers_train(name, classifier_orig):
             assert_greater(accuracy_score(y, y_pred), 0.83)
 
         # raises error on malformed input for predict
-        with assert_raises(ValueError, msg="The classifier {} does not"
-                           " raise an error when the number of features "
-                           "in predict is different from the number of"
-                           " features in fit.".format(name)):
-            classifier.predict(X.T)
+        if not is_pairwise(classifier):
+            with assert_raises(ValueError, msg="The classifier {} does not"
```

Comment:
Why doesn't this apply in the pairwise case? Surely in the pairwise case the estimator still needs to raise an error if the argument to predict isn't a matrix of shape `(n_test_samples, n_training_samples)`